src/adapter/src/catalog.rs

```diff
@@ -2849,6 +2850,114 @@ impl Catalog {
                         )?;
                     }
                 }
+                Op::RenameSchema {
+                    database_spec,
+                    schema_spec,
+                    new_name,
+                } => {
+                    let ResolvedDatabaseSpecifier::Id(database_id) = &database_spec else {
+                        let err = anyhow::anyhow!("Cannot rename schemas in the ambient database");
+                        return Err(AdapterError::Unstructured(err));
```

Comment:
We should use the `ReadOnlySchema` error here. You can see an example in `Op::AlterSource`.

Reply:
Refactored this away